### PR TITLE
refactor: integrate new cu

### PR DESCRIPTION
**Pull request by pawanpaudel93**
## Summary

This PR updates the default AO CU to `https://gateway.ar`. Also, this PR makes sure `syncAoTokens` is run only once even if a user goes to `View all`, goes back and again goes back to `View all`.

### CODE REVIEW

File: src/tokens/aoTokens/ao.ts [hunk 1]
```diff
@@ -144,7 +145,7 @@ export function useAoTokens(): [TokenInfoWithBalance[], boolean] {
               const balance = Number(
                 await timeoutPromise(
                   (async () => {
-                    const aoToken = await Token(id);
+                    const aoToken = await Token(id, null, ao);
```
Comment (pawanpaudel93) [inline src/tokens/aoTokens/ao.ts:148]: @7i7o Should we not use `Token` here and directly fetch balance as this first fetches the token info and then the balance ? Can we directly fetch the balance and use the `Denomination` saved on the storage ?

Comment (7i7o) [reply src/tokens/aoTokens/ao.ts:148]: We could, but when could we update the Denomination if someone changes that?

Comment (pawanpaudel93) [reply src/tokens/aoTokens/ao.ts:148]: Yes, that can be the issue if we use the stored one and if Denomination of a token changes. We can leave it as it is.

**Comment by 7i7o** [comment]
Closing for this was solved in subsequent PRs

**Pull request closed without merging**